core/src/main/scala/org/apache/spark/ui/jobs/AllJobsPage.scala

```diff
@@ -295,6 +296,12 @@ private[ui] class AllJobsPage(parent: JobsTab) extends WebUIPage("") {
                 <strong>Total Uptime: </strong>
                 {UIUtils.formatDuration(System.currentTimeMillis() - startTime)}
               </li>
+            } else {
```

Comment:
I think that's probably fine, though the duplication between these branches can be reduced. Only the source of the end time differs. Thoughts @sarutak ?
This is a dumb question but will this change actually affect the view from the history server?